src/NuGet.Clients/NuGet.SolutionRestoreManager/ISolutionRestoreChecker.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using NuGet.Commands;
+using NuGet.ProjectModel;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// An up to date checker for a solution.
+    /// </summary>
+    public interface ISolutionRestoreChecker
+    {
+        /// <summary>
+        /// Given the current dependency graph spec, perform a fast up to date check and return the dirty projects.
```

Comment:
Nit: make lines in `<summary>` shorter, and `<remarks>` multiline, to make it easier to read in the diff :)